### DISCUSSION TITLE
Весь трафик идет через AntiZapret

### DISCUSSION THREAD

**Discussion by Selys-u** [xtrime-ru/antizapret-vpn-docker#42] [Q&A] [answered]
Доброго времени суток! Подскажите, по какой причине весь трафик идет через туннель?

Роутер Keenetic, добавил профиль, настроил в интернет фильтрах DNS, отдельно в профиле системный и в профиле AntiZapret, профиль фильтрации по умолчанию - AntiZapret, в клиентском конфиге добавил route на DNS из профиля AntiZapret (62.76.62.76 и 9.9.9.10). Тот же 2ip.ru показывает ip хоста с Антизапретом

Контейнер запущен Adguard, никакие параметры не менял в docker-compose.adguard.yml

**Comment by xtrime-ru** [comment]
Давайте начнем с тупых вопросов :)

1. Почему считаете что весь трафик идет через тунель? 
2. Что выдает `nslookup ya.ru  192.168.100.1` (udp) или `nslookup ya.ru  192.168.104.1` (tcp)
3. Что выдает traceroute до заблокированного и незаблокированного сайтов?

**Comment by Selys-u** [comment]
Я знал, что будут вопросы такие =) 
Простите, что сразу не предоставил.
Тот же speedtest указывает на ip vps и скорость мизерная(

1. Потому что все сайты, которые показывают ip адрес внешний, отображают адрес моего VPS, даже сайты, которые не блокируются в РФ

2.  `nslookup ya.ru  192.168.100.1`
>nslookup ya.ru  192.168.100.1
>╤хЁтхЁ:  UnKnown
>Address:  192.168.100.1

>Не заслуживающий доверия ответ:
>╚ь :     ya.ru
>Addresses:  10.224.2.93
>         10.224.2.94
>         10.224.2.92

3. `tracert -d ya.ru`
>Трассировка маршрута к ya.ru [10.224.2.92]
>с максимальным числом прыжков 30:

>  1     2 ms     2 ms     2 ms  192.168.0.1
>  2    45 ms    46 ms    47 ms  10.224.2.92
>  3    46 ms    46 ms    45 ms  10.224.2.92
>  4    48 ms    54 ms    53 ms  10.224.2.92
>  5    47 ms    54 ms    47 ms  10.224.2.92
>  6    46 ms    46 ms    47 ms  10.224.2.92
>  7    82 ms    90 ms    82 ms  10.224.2.92
>  8    81 ms    84 ms    81 ms  10.224.2.92

**Reply by xtrime-ru** [reply]
Просьба выполнить команду на хосте: 
```shell
docker exec -it antizapret-vpn cat /etc/knot-resolver/knot-aliases-alt.conf | tail -n3
```

**Comment by Selys-u** [comment]
Выполнил:
>}
>regex_blocked = '()'
>regex_allowed = '^$'

**Reply by xtrime-ru** [reply]
Вот и проблема! 
`regex_blocked = '()'`
А должно быть 
`regex_blocked = '^$'`

Версия контейнера самая последняя? Что в файле `./config/exclude-regex-custom.txt` ?

**Reply by Selys-u** [reply]
Учитывая, что вчера только установил, предполагаю, что последняя =)

В файле пусто:  `cat ./config/exclude-regex-custom.txt` ничего не выдает. 
Хм, я попробовал туда 2ip.ru (^2ip\.ru$ из Documentation) вписать, проверить просто, как работает, но потом удалил

**Reply by Selys-u** [reply]
После того, как удалил, много раз уже ребутал контейнер

**Reply by xtrime-ru** [reply]
Проблема в том, что в файле ./config/exclude-regex-custom.txt скорее всего пробелы или что то подобное.
Речь именно про exclude файл. 2ip надо добавлять в include файл. 

Для того что бы починить достаточно сделать
```shell
rm ./config/exclude-regex-custom.txt && docker compose restart
```

Я проверю, как там фильтрация пустых строк работает, что бы небыло таких проблем.

**Reply by Selys-u** [reply]
Суть понял, проверил файл, там пусто, абсолютно. На всякий случай выполнил
`rm ./config/exclude-regex-custom.txt && docker compose restart`
но ничего не поменялось.
 Вывод от
`docker exec -it antizapret-vpn cat /etc/knot-resolver/knot-aliases-alt.conf | tail -n3`
>}
>regex_blocked = '()'
>regex_allowed = '^$'

**Reply by Selys-u** [reply]
Может, в самом контейнере почистить `knot-aliases-alt.conf` и посмотреть заполнится ли он или нет?

**Reply by xtrime-ru** [reply, marked as answer]
Сделал фикс, просьба обновить контейнер: 
https://github.com/xtrime-ru/antizapret-vpn-docker/commit/b00efeaf5be31ed43d26d9d6869c4f5d656df979

```shell
docker compose pull && docker compose up -d
```

**Reply by Selys-u** [reply]
Теперь вывод следующий:
>}
>regex_blocked = ''
>regex_allowed = ''

**Reply by Selys-u** [reply]
Всё заработало должным образом. Спасибо огромное, очень сильно выручили

**Reply by Selys-u** [reply]
Интересно, а у всех там было и они не обращали внимания? Или мне так повезло?

**Reply by xtrime-ru** [reply]
Да, все хорошо. Я сменил значение по умолчанию на пустую строку. 
Это так же позволило оптимизировать алгоритм.

**Reply by xtrime-ru** [reply]
> Или мне так повезло?

Как я уже написал проблема возникала, если в файлах *-regex-custom.txt были только пустые строки. По умолчанию файлы не содержали вообще никаких строки и все было хорошо. Так же если туда добавить хотя бы одну не пустую строку, то все так же работало корректно. 

Но баг был, так что спасибо за сообщение!

**Reply by Selys-u** [reply]
И вам спасибо за оперативность и за ваш проект в целом!
